src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMapArchiver.java

```diff
@@ -25,20 +27,41 @@
 {
     private static final Logger logger = LoggerFactory.getLogger(CountryBoundaryMapArchiver.class);
 
-    private static final Switch<java.io.File> SHAPE_FILE = new Switch<>("shp",
-            "path to the shape file", java.io.File::new, Optionality.OPTIONAL);
+    // Parameters
+    private static final Switch<File> SHAPE_FILE = new Switch<>("shp", "path to the shape file",
+            File::new, Optionality.OPTIONAL);
     private static final Switch<Atlas> ATLAS = new Switch<>("atlas",
             "path to the atlas file containing boundaries",
             path -> PackedAtlas.load(new File(path)), Optionality.OPTIONAL);
-    private static final Switch<File> OUT = new Switch<>("out", "The output file format",
+    private static final Switch<File> OUTPUT = new Switch<>("out", "The output file format",
             path -> new File(path));
+    public static final Switch<StringList> COUNTRIES = new Switch<>("countries",
+            "Comma separated list of countries to be included in the boundary file",
+            value -> StringList.split(value, ","), Optionality.OPTIONAL);
     private static final Switch<Rectangle> BOUNDS = new Switch<>("bounds", "The bounds",
             rectangle -> Rectangle.forString(rectangle), Optionality.OPTIONAL,
             Rectangle.MAXIMUM.toCompactString());
     private static final Switch<Boolean> CREATE_SPATIAL_INDEX = new Switch<>("createSpatialIndex",
-            "Default true, performance optimization to create and serialize a spatial index.",
+            "Indicator whether to create a spatial grid index and include that in the output.",
             Boolean::parseBoolean, Optionality.OPTIONAL, Boolean.FALSE.toString());
 
+    private static final Function<Iterable<String>, Predicate<String>> COUNTRY_FILTER_GENERATOR = countryList ->
+    {
+        return countryLine ->
+        {
+            final String readCountryName = StringList
+                    .split(countryLine, CountryBoundaryMap.COUNTRY_BOUNDARY_DELIMITER).get(0);
+            for (final String countryName : countryList)
```

Comment:
Instead of for loop how about
``` return countryList.stream().anyMatch(countryName -> countryName.equals(readCountryName))```

Reply:
Good point, I'll look into a way to improve this block of code.